dockstore/dockstore-ui2
# Display multiple authors in search

**Description**
Corresponding webservice PR: https://github.com/dockstore/dockstore/pull/5461

This PR displays multiple authors in the search page. ORCID authors have a link to their ORCID page.

I've removed most usages of the Entry `author` field. The list entries page like this [one](https://qa.dockstore.org/workflows) still use the Entry `author`. Removing that would be a little more involved because it uses the `author` as sorting query parameter to the `allPublished` endpoint. I created a ticket https://github.com/dockstore/dockstore/pull/5461 to remove this usage of the `author` field.

Misc changes while I was in the area:
- Added a tooltip for the topics on the search page so that users can still read the full topic even if they're truncated
- Some facet names were camel case, so I added spaces

Screenshots:
![Screenshot 2023-04-14 at 17-29-04 Dockstore Search](https://user-images.githubusercontent.com/25287123/232162559-07987851-6c90-4aa2-af59-d673fe96fa9c.png)

![Screenshot 2023-04-14 at 17-51-57 Dockstore Search](https://user-images.githubusercontent.com/25287123/232162570-9da1ebee-1b8f-498f-8a0a-6ff694a52f27.png)


**Review Instructions**
- Create a GitHub app workflow with a .dockstore.yml that **doesn't** contain authors. The descriptor of the workflow should contain an author.
- Publish the workflow
- Verify that the search page displays the descriptor author for this workflow
- Add two or more authors to the .dockstore.yml (we want multiple authors)
- Verify that the search page displays the .dockstore.yml authors for this workflow. There should be a search facet for each author.

**Issue**
[SEAB-3629](https://ucsc-cgl.atlassian.net/browse/SEAB-3629)

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!



[SEAB-3629]: https://ucsc-cgl.atlassian.net/browse/SEAB-3629?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
